server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexRepository.scala

```diff
@@ -51,56 +51,119 @@ trait ElasticsearchRepository {
     * @param tokens address tokens
     * @return Future with found PAF addresses and the maximum score
     */
-  def queryNagAddresses(tokens: AddressTokens) : Future[NationalAddressGazetteerAddresses]
+  def queryNagAddresses(tokens: Seq[CrfTokenResult]) : Future[NationalAddressGazetteerAddresses]
 }
 
 @Singleton
-class AddressIndexRepository @Inject()(conf : AddressIndexConfigModule, elasticClientProvider: ElasticClientProvider)(implicit ec: ExecutionContext) extends ElasticsearchRepository {
+class AddressIndexRepository @Inject()(
+  conf: AddressIndexConfigModule,
+  elasticClientProvider: ElasticClientProvider
+)(implicit ec: ExecutionContext) extends ElasticsearchRepository {
 
   private val esConf = conf.config.elasticSearch
   private val pafIndex = esConf.indexes.pafIndex
   private val nagIndex = esConf.indexes.nagIndex
   val client: ElasticClient = elasticClientProvider.client
+  private val logger = Logger("AddressIndexRepository")
+
+  def queryPafUprn(uprn: String): Future[Option[PostcodeAddressFileAddress]] = {
+    client execute {
+      search in pafIndex query { termQuery("uprn", uprn) }
+    } map(_.as[PostcodeAddressFileAddress].headOption)
+  }
+
+  def queryNagUprn(uprn: String): Future[Option[NationalAddressGazetteerAddress]] = {
+    client execute {
+      search in nagIndex query { termQuery("uprn", uprn) }
+    } map(_.as[NationalAddressGazetteerAddress].headOption)
+  }
 
-  def queryPafUprn(uprn: String): Future[Option[PostcodeAddressFileAddress]] = client.execute{
-    search in pafIndex query { termQuery("uprn", uprn) }
-  }.map(_.as[PostcodeAddressFileAddress].headOption)
+  private def getTokenValue(token: Token, tokens: Seq[CrfTokenResult]): String = {
+    tokens.filter(_.label == token).map(_.value).mkString(" ")
+  }
 
-  def queryNagUprn(uprn: String): Future[Option[NationalAddressGazetteerAddress]] = client.execute{
-    search in nagIndex query { termQuery("uprn", uprn) }
-  }.map(_.as[NationalAddressGazetteerAddress].headOption)
+  private def collapseTokens(tokens: Seq[CrfTokenResult]): Map[Token, Input] = {
+    tokens.groupBy(_.label).map(e => e._1 -> e._2.map(_.value).mkString(" "))
+  }
+
+  def tokensToMatchQueries(tokens: Seq[CrfTokenResult], tokenFieldMap: Map[Token, String]): Iterable[QueryDefinition] = {
+    collapseTokens(tokens) flatMap { case (tkn, input) =>
```

Comment:
I'm curious to see where "scala val pref is to remove vowels" comes from, I couldn't google the answer. If it's not used, it could be replaced with `_`.
I mean if `token` is replaced by `tkn`, why leave `input` as it is instead of making it `npt`? Or I'm missing something?